### PR TITLE
Feature/jameszhang878/side nav component

### LINKED ISSUE #15: Side Navigation Component
Make a generalized side navigation component that renders content on the right based on which page is selected on the left. 

Screen Shot 2022-01-14 at 11 55 11 AM

This component comes up in the Wish Granting and Manage tabs in the MVP. Because of this, we want to be able to reuse it between pages and with arbitrary tab names and content. One possible way to do this would be to pass in an array of object in the following form 
```
{
    tab_id: ___,
    tab_name: ___,
    tab_content: <content represented in jsx>
}
```

~~This should be mobile compatible (ask designers during 1/14 meeting about how to make this possible, maybe floating / sticky nav button that expands a drawer).~~

### PR DESCRIPTION
**Pull request by JamesZhang878**
### Administrative Info
[GitHub Issue #15 ]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added a side navigation component that can take in information to populate the component. User can navigate between page components passed in through props. Also created blank js files for the components that should be rendered on the Wish Granting page.

### Testing
How did you confirm your changes worked? 
- I ran the react app to make sure that it worked.

### Confirmation of Change 

https://user-images.githubusercontent.com/73041370/150609098-014638db-7a38-4933-97d6-e774b134e400.mov

### CODE REVIEW

**Review by PatrickBrown1 — APPROVED** [review]
I went ahead and added the box shadow to the sidenav component. Otherwise lgtm. We should think about what should dictate the height of the component. Currently, height: 100% doesn't fill out the whole page. It will if the child page component (what shows on the right) fills the whole page, which is what makes the most sense to me (and thus, is why I am approving).

File: frontend/src/components/SideNav.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  // Changes the content displayed, and the tab that gets highlighted.
+  const changeTab = (id) => {
+    setDisplayed(tabs[id].tab_content);
+    setSelected(id);
+  };
+
+  return (
+    <div className="side_nav_layout">
+      <div className="side_nav_buttons">
+        {tabs.map((tab) => (
```
Comment (Cubified) [inline frontend/src/components/SideNav.js:31]: The `tab_id` field on each tab isn't strictly necessary, you could do this instead:

```js
{tabs.map((tab, tab_id) => (
```

The extra parameter `tab_id` stores the index of the tab, so you don't have to store that value manually.

---

Also React will complain if you don't give every button a unique `key` property, so I'd just add something like `key={tab.tab_name}` right below this.

File: frontend/src/pages/WishGrantingPage.js [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+import WishDiscovery from "../components/WishDiscovery";
+import WishPlanning from "../components/WishPlanning";
+import WishEnhancements from "../components/WishEnhancements";
+import WishReveal from "../components/WishReveal";
+import WishCloseout from "../components/WishCloseout";
 
 function WishGrantingPage() {
+  const tabs = [
+    {
+      tab_id: 0,
```
Comment (Cubified) [inline frontend/src/pages/WishGrantingPage.js:12]: (Same thing as the comment from before, you don't need to put a tab_id if you give `map` an extra parameter)

**Review by Cubified — APPROVED** [review]
I suggested two minor changes that are both in `SideNav.js`, but other than that LGTM.

**Pull request merged**